src/main/java/org/jax/mvar/utility/App.java

```diff
@@ -70,75 +76,114 @@ private static Map<String, Object> cmdArgsParser(String[] args) {
                     case "-check_canon":
                         arguments.put("check_canon", true);
                         break;
+                    case "-lifted":
+                        arguments.put("lifted", true);
+                        break;
                     case "-imputed":
                         arguments.put("imputed", Byte.valueOf(args[i+1]));
                         break;
                     case "-header_path":
                         arguments.put("header_path", args[i+1]);
                         break;
+                    case "-reference":
+                        arguments.put("reference", args[i+1]);
+                        break;
                     default:
-                        throw new IllegalStateException("Unexpected parameter: " + args[0]);
+                        throw new IllegalStateException("Unexpected parameter: " + args[i]);
                 }
             }
         }
         return arguments;
     }
 
     public static void main(String[] args) {
```

Comment:
- Consider adding documentation for the parameters. 
- Consider refactoring this method as the complexity seems high given the length and logic in the method